### PR TITLE
[KOGITO-3270] - Accepting external ConfigMaps to be mounted in a give…

### PR DESCRIPTION
**Pull request by ricardozanini**
…n KogitoService

See: https://issues.redhat.com/browse/KOGITO-3270

In this PR we introduce a new attribute in the Kogito services interface, which can be informed to tell the operator to not create a configMap, but to use the one given in advance. This way users can configure the application upfront using an external process (like a CI) before deploying a given Kogito service in the namespace.

For example, one can create a configMap with:

```shell
$ oc create configmap mycustom-configmap --from-file=application.properties 
```

And then deploy the KogitoRuntime with:

```yaml
apiVersion: app.kiegroup.org/v1alpha1
kind: KogitoRuntime
metadata:
  name: process-quarkus-example
spec:
  propertiesConfigMap: mycustom-configmap
  replicas: 1
  image:
    domain: quay.io
    namespace: kiegroup
    name: process-quarkus-example
```

The same applies for any other supporting service like Management Console or Data Index.

Every property from the given CM, manually inserted properties, or properties manipulated by the operator will be in this CM and managed by the operator.

Note: I'm adding a RELEASE_NOTES.md file to be used later by our CI to create the Release Notes in the GH page. Please update this file from now on when sending a PR that have a feature/bug fix that needs to be included in the release notes.

Signed-off-by: Ricardo Zanini <zanini@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Review by spolti — APPROVED** [review]

**Comment by vaibhavjainwiz** [comment]
@ricardozanini 

I have some doubts, could you please help me to clarify them

- User could provide more then one config map.

- User could creates config map from file as well as from literal.

- With new approch in KOGITO-3039, kogito operator will create seperate config map for each kogitoInfra object. I think its better to create seperate config map for kogito-operator managed properties. With this we could easy maintain config map ownership and we would not made any changes with user managed config map.

- Could we let user to provide its volume/volume mount if he is creating config map from file or we have to take extra input from user whether he create config map from file or literal? I think it will be better from user usablity point of view if we could provide two seperate property like `configMapFromLiteral` and `configMapFromFile`. With this user don't need to provide volume/volume mount details.

**Comment by ricardozanini** [comment]
> @ricardozanini
> 
> I have some doubts, could you please help me to clarify them
> 
>     * User could provide more then one config map.

No, they can't. This approach is to just configure their application based on Quarkus or Spring runtimes. The configMap is mounted in the `config` directory, which is read by the application in runtime.

>     * User could creates config map from file as well as from literal.

This configMap supposed to be mounted as an `application.properties` file. Whatever the way they create the CM, if it has the `application.properties` key will work.

>     * With new approch in KOGITO-3039, kogito operator will create seperate config map for each kogitoInfra object. I think its better to create seperate config map for kogito-operator managed properties. With this we could easy maintain config map ownership and we would not made any changes with user managed config map.

We shouldn't create separated CMs since all the properties should be in the `application.properties` file mounted in the pod. We can however, create one CM for each infra for our control, in our format. Then we can merge all the properties in the one which will be mounted in the `config` directory.

>     * Could we let user to provide its volume/volume mount if he is creating config map from file or we have to take extra input from user whether he create config map from file or literal? I think it will be better from user usablity point of view if we could provide two seperate property like `configMapFromLiteral` and `configMapFromFile`. With this user don't need to provide volume/volume mount details.

I don't think we need to open that much, users can create/map their on CMs/keys using the `envs` attribute. About volume/volume mounts, this can be another JIRA/conversation if they need additional files being mounted. We can let this discussion to the future, since all files needed by the application are handled: protobuf and properties. This can change of course.

The main motivation here is just for the user to inform a custom application.properties to the operator before the deployment. This will cover CI/CD process where you have a Consul/Netflix/Spring properties vault which you mount then during this process and set that CM to the Kogito CR.

**Review by vaibhavjainwiz — APPROVED** [review]
LGTM

**Review by sutaakar — APPROVED** [review]
Looks good to me, good job.

**Comment by sutaakar** [comment]
@ricardozanini Looks like the property is currently not provided in CLI. DO you plan to add CLI support?

**Comment by ricardozanini** [comment]
@sutaakar yes indeed I have to add a flag in the deployment command. Will do it today.

**Comment by ricardozanini** [comment]
@sutaakar please take a look again. I've added a new feature to the CLI where an user could give a local file to be set as a property file to the application. Do you mind verifying it? :-)

Many thanks!

File: cmd/kogito/command/service/config.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure/services"
+	"io/ioutil"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	configMapPrefix     = "-custom-properties"
```
Comment (sutaakar) [inline cmd/kogito/command/service/config.go:29]:
```suggestion
	configMapPrefix     = "custom-properties"
```
Current implementation creates double dash: `example-quarkus--custom-properties`

**Review by sutaakar — APPROVED** [review]
Looks good to me, works as expected.
One minor suggestion above.

File: cmd/kogito/command/service/config.go (hunk 1 above)
Comment (sutaakar) [reply cmd/kogito/command/service/config.go:29]: Also could be called `configMapSuffix` to be precise :)

Comment (ricardozanini) [reply cmd/kogito/command/service/config.go]: LOL! I'm pretty dumb. Gonna fix ASAP. Many thanks @sutaakar 

Have you had the opportunity to verify it?

Comment (sutaakar) [reply cmd/kogito/command/service/config.go]: yes, works as expected

**Review by spolti — APPROVED** [review]

**Comment by ricardozanini** [comment]
@spolti AWS is down

**Pull request merged**